packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -5439,6 +5446,15 @@ class NavigatorState extends State<Navigator> with TickerProviderStateMixin, Res
   /// Notifies its listeners if the value of [userGestureInProgress] changes.
   final ValueNotifier<bool> userGestureInProgressNotifier = ValueNotifier<bool>(false);
 
+  /// Notifies its listeners if there is a delegate transition from the top route.
+  final ValueNotifier<DelegatedTransitionBuilder?> delegateTransitionBuilderNotifier = ValueNotifier<Widget Function(BuildContext context, Widget? child, Animation<double> animation)?>(null);
+
+  /// Sets the delegate transition.
+  set delegateTransitionBuilder(DelegatedTransitionBuilder? builder) => delegateTransitionBuilderNotifier.value = builder;
+
+  /// Gets the delegate transition.
+  DelegatedTransitionBuilder? get delegateTransitionBuilder => delegateTransitionBuilderNotifier.value;
```

Comment:
This might be better as a derived value instead of being saved. Something like:

```dart
  DelegatedTransitionBuilder? get delegateTransitionBuilder {
    for (final _RouteEntry routeEntry in _history) {
      if (routeEntry.route is! ModalRoute) {
        continue;
      }
      final ModalRoute<dynamic> modalRoute = routeEntry.route as ModalRoute<dynamic>;
      if (modalRoute.delegatedTransition == null) {
        continue;
      }
      return modalRoute.delegatedTransition;
    }
    return null;
  }
```

Reply:
Actually, after talking this over offline, I think we don't need to iterate all routes. Just check the top route for a delegatedTransition.

Pushing another route onto the root navigator after an MBS route is already on there should not use the MBS delegatedTransition. So we're defining this feature to only allow changing the transition of the top route (the outgoing route when the MBS-style route is pushed).